assets/dev/js/editor/components/template-library/views/template/cloud.js

```diff
@@ -27,6 +28,66 @@ TemplateLibraryTemplateCloudView = TemplateLibraryTemplateLocalView.extend( {
 		} );
 	},
 
+	modelEvents: _.extend( {}, TemplateLibraryTemplateLocalView.prototype.modelEvents, {
+		'change:preview_url': 'onPreviewUrlChange',
+	} ),
+
+	onRender() {
+		const previewUrl = this.model.get( 'preview_url' );
+
+		if ( this.needGeneratePreview() ) {
+			this.iframe = elementor.templates.layout.createScreenshotIframe( this.model.get( 'generate_preview_url' ) );
+			this.isGeneratingPreview = true;
+		}
+
+		if ( previewUrl ) {
+			this.updatePreviewImgStyle();
+		}
+	},
+
+	onPreviewUrlChange() {
+		const isCloudSource = 'cloud' === this.model.get( 'source' );
+		const previewUrl = this.model.get( 'preview_url' );
+		this.isGeneratingPreview = false;
+
+		if ( isCloudSource && previewUrl ) {
+			this.ui.previewImg.attr( 'src', previewUrl );
+			this.updatePreviewImgStyle();
+			this.model.set( 'generate_preview_url', null );
+			this.iframe.remove();
+		}
+	},
+
+	updatePreviewImgStyle() {
+		this.ui.previewImg.css( 'object-fit', 'contain' );
+	},
+
+	needGeneratePreview() {
+		const view = elementor.templates.getViewSelection();
```

Comment:
I liked the previous syntax more :)

```
return 'grid' !== elementor.templates.getViewSelection() &&
    'FOLDER' === this.model.get( 'subType' )
    etc
```
Unless you had a specific reason for writing it like this.